lib/streamlit/runtime/state/widgets.py

```diff
@@ -148,12 +151,16 @@ def register_widget(
         For both paths a widget return value is provided, allowing the widgets
         to be used in a non-streamlit setting.
     """
-    widget_id = compute_widget_id(element_type, element_proto, user_key)
-    element_proto.id = widget_id
+    if not element_proto.id:
+        widget_id = compute_widget_id(element_type, element_proto, user_key)
+        element_proto.id = widget_id
+        LOGGER.warn(
+            f"Registering a widget with no id, falling back to generating from proto. {widget_id=}"
+        )
```

Comment:
Is this something that we'll want to get rid of before checking this code into `develop`, or are there still instances of us falling back to hashing protos to generate widget IDs? Might want to leave a small comment describing when we still need the fallback if so

Reply:
I included it out of caution, in case I had missed something in some edge case, but everything works even replacing it with an assert that id is set, so it shouldn't be needed anymore.